Week5/week 5.txt

```diff
@@ -0,0 +1,41 @@
+import csv
+with open("c:\\sqlite3\\student.csv","w",newline="")as stud:
+    write=csv.writer(stud)
+    header=['SId','SName','City','Contact']
+    write.writerow(header)
+    rows=[
+          [1,'savan','Navsari',123456789],
+          [2,'jay','Surat',123456788],
+          [3,'hiren','Bardoli',123456787],
+          [4,'Radha','Bardoli',123456786],
+          [5,'Kishan','Surat',123456784]
+         ]
+    write.writerows(rows)
+    print('Record Inserted successful')
+   
+ #5 records insert using user input
+
+    l=[]
+    for record in range(5):
+        n=int(input('Enter Student ID:'))
+        name=input('Enter Student Name: ')
+        city=input('Enter City Name: ')
+        contact=int(input('Enter Contact Number:'))
+        h=[n,name,city,contact]
+        l.append(h)
+    write.writerows(l)
+    print('Record Inserted successful')
+
+#Read this file using csv module and print it. 
+
+with open("c://sqlite3//student.csv","r",newline="")as stud:
+    read=csv.reader(stud)
+    for i in read:
+        print(i)
+
+#Reading and writting must be perform using python script.
+
+with open("c:\\sqlite3\\student.csv","r")as stud:
+    r=csv.DictReader(stud)

```

Comment:
whats difference between DictReader() and Writerows()?

Reply:
DictReader() and writerows() are functions commonly used when working with CSV (Comma-Separated Values) files in Python, typically through the csv module.


writerows():

writerows() is used to write data from Python data structures (such as lists of dictionaries) into a CSV file.
It expects an iterable (e.g., a list of dictionaries), where each element represents a row of data.
The data is written to the CSV file in a structured format.

DictReader():

DictReader() is used to read data from a CSV file into a Python data structure.
It creates a reader object that interprets the CSV file's rows as dictionaries, where each column header becomes a dictionary key, and the corresponding value is the data in the respective column.
You can then iterate through the rows using a for loop and access data using dictionary keys.|